spinoso-string/src/center.rs

```diff
@@ -0,0 +1,174 @@
+use core::fmt;
+use core::iter::{Cycle, FusedIterator, Take};
+use core::slice;
+
+use crate::chars::Chars;
+
+/// Error returned when failing to construct a [`Center`] iterator.
+///
+/// This error is returned from [`String::center`]. See its documentation for
+/// more detail.
+///
+/// This error corresponds to the [Ruby `ArgumentError` Exception class].
+///
+/// When the **std** feature of `spinoso-string` is enabled, this struct
+/// implements [`std::error::Error`].
+///
+/// [`String::center`]: crate::String::center
+/// [Ruby `ArgumentError` Exception class]: https://ruby-doc.org/core-2.6.3/ArgumentError.html
+/// [`std::error::Error`]: https://doc.rust-lang.org/std/error/trait.Error.html
+#[allow(clippy::module_name_repetitions)]
```

Comment:
You can imagine why this is needed. It could be renamed `Error`, but it feels less descriptive.

If it was to be renamed to just Error, then we'd need to do `pub use center::Error as CenterError` in lib.rs to keep the public api for spinoso-string the same. Not sure how idiomatic that is though.